deviations/views.py

```diff
@@ -2,13 +2,59 @@
 from django.db import IntegrityError
 from django.shortcuts import get_object_or_404
 from django.utils.translation import ugettext_lazy as _
+from userprofile.models import UserProfile
 
 from course.viewbase import CourseInstanceBaseView, CourseInstanceMixin
 from lib.viewbase import BaseFormView, BaseRedirectView
 from authorization.permissions import ACCESS
-from .forms import DeadlineRuleDeviationForm
+from .forms import DeadlineRuleDeviationForm, RemoveDeadlineRuleDeviationForm
 from .models import DeadlineRuleDeviation
 from exercise.models import BaseExercise
+from itertools import chain
+
+
+class OverrideDeadlinesView(CourseInstanceMixin, BaseFormView):
+    access_mode = ACCESS.TEACHER
+    template_name = "deviations/override.html"
+    form_class = DeadlineRuleDeviationForm
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["instance"] = self.instance
+        return kwargs
+
+    def get_success_url(self):
+        return self.instance.get_url("deviations-list-dl")
+
+    def get_common_objects(self):
+        super().get_common_objects()
+        deviation_list = []
+        for e, s in self.request.session['already_have_deviation']:
+            deviation_list += [DeadlineRuleDeviation.objects.get(
+                    exercise=BaseExercise.objects.get(id=e),
+                    submitter=UserProfile.objects.get(id=s),
+                    exercise__course_module__course_instance=self.instance)]
+
+        self.deviations = deviation_list
+        self.without_late_penalty = self.request.session['without_late_penalty']
+        self.minutes = self.request.session['minutes']
+        self.note("deviations", "without_late_penalty", "minutes")
+
+    def form_valid(self, form):
+
+        deviation_list = self.request.POST.getlist('override')
+        minutes = self.request.session['minutes']
+        without_late_penalty = self.request.session['without_late_penalty']
+        for string in deviation_list:
+            e, s = string.split('-')
+            DeadlineRuleDeviation.objects.filter(
+                exercise=BaseExercise.objects.get(id=e),
+                submitter=UserProfile.objects.get(id=s)).update(
+                extra_minutes=minutes,
+                without_late_penalty=without_late_penalty)
+
+        del self.request.session['already_have_deviation']
```

Comment:
Shouldn't all session keys (the ones that are used for this deviation page) be deleted at this stage?